tests/python/test_python_frontend.py

```diff
@@ -4280,3 +4282,95 @@ def fusion_func(fd: FusionDefinition):
             ),
         ):
             nvf_out = fd.execute([tensor_inp, 2.0 + 1.0j])
+
+    def test_repro_script_generation(self):
+        expected_str = """
+# CUDA devices:
+#  0: NVIDIA A100 80GB PCIe
+# torch version: 2.5.0a0+git25d5a81
+# cuda version: 12.6
+# nvfuser version: 0.2.10+gitb2e3f89
+import torch
+from nvfuser import FusionDefinition, DataType
+
+def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
+    T0 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Float, is_cpu=False, stride_order=[1, 0])
+    T1 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=DataType.Float, is_cpu=False, stride_order=[1, 0])
+    T2 = fd.ops.mul(T0, T1)
+    fd.add_output(T2)
+
+with FusionDefinition() as fd:
+    nvfuser_fusion_id0(fd)
+
+inputs = [
+    torch.randn(16, dtype=torch.float32, device='cuda:0').as_strided((4, 4), (4, 1)),
+    torch.randn(16, dtype=torch.float32, device='cuda:0').as_strided((4, 4), (4, 1)),
+]
+fd.execute(inputs)
+"""
+
+        inputs = [
+            torch.randn(4, 4, device="cuda:0"),
+            torch.randn(4, 4, device="cuda:0"),
+        ]
+
+        def fusion_func(fd: FusionDefinition):
+            t0 = fd.from_pytorch(inputs[0])
+            t1 = fd.from_pytorch(inputs[1])
+            t2 = fd.ops.mul(t0, t1)
+            fd.add_output(t2)
+
+        with FusionDefinition() as fd:
+            fusion_func(fd)
+
+        # Strip white space before and after script
+        expected_str = expected_str.strip()
+        expected_lines = expected_str.split("\n")
+
+        # Test fd.execute(inputs, print_repro=True)
+        old_stdout = sys.stdout
+        test_stdout = sys.stdout = io.StringIO()
+
+        fd.execute(inputs, print_repro=True)
+
+        sys.stdout = old_stdout
+
+        comp_str = test_stdout.getvalue()
+        comp_str = comp_str.strip()
+        comp_lines = comp_str.split("\n")
+
+        assert len(expected_lines) == len(
+            comp_lines
+        ), "Reproduction script does not have the expected number of lines! Expected: {} Actual: {}".format(
+            len(expected_lines), len(comp_lines)
+        )
+        # Don't compare the first 5 lines as they may be device and cuda specific
```

Comment:
Nit: could instead use `re.match` in order to handle the case where multiple devices are available, in which case there will be more than 5 comment lines.

Reply:
further nitpick, might as well do ast.parse -> ast.dump and then just compare that. But that might be an overkill, and that still won't save us from variable renaming and stuff like that....

Reply:
I think I'm just saying that I like Jacob's suggestion above.

Reply:
I fixed this by not including the starter header in the comparison by looking for comments.